spec/models/evidence/forms/income_spec.rb

```diff
@@ -15,45 +16,67 @@
   end
 
   describe 'validation' do
+    before do
+      form.update_attributes(params)
+    end
+
+    subject { form.valid? }
+
+    context 'when the income is above 0' do
+      let(:income) { '100' }
+
+      it { is_expected.to be true }
+    end
+
     context 'when the income is 0' do
-      let(:amount) { '0' }
+      let(:income) { '0' }
 
-      it { expect(subject.valid?).to be true }
+      it { is_expected.to be true }
     end
 
     context 'when the income is negative' do
-      let(:amount) { '-1' }
+      let(:income) { '-1' }
 
-      it { expect(subject.valid?).to be false }
+      it { is_expected.to be false }
     end
 
     context 'when the income is string' do
-      let(:amount) { 'a string' }
+      let(:income) { 'a string' }
 
-      it 'does not pass validation' do
-        expect(subject.valid?).to be false
-      end
+      it { is_expected.to be false }
     end
 
     context 'when the income is blank' do
-      let(:amount) { '' }
+      let(:income) { '' }
 
-      it 'does not pass validation' do
-        expect(subject.valid?).to be false
-      end
+      it { is_expected.to be false }
     end
   end
 
   describe '#save' do
+    let(:evidence) { create :evidence_check }
+    let(:params) { { income: '500.5' } }
+    let(:income_calculation_result) { { outcome: 'part', amount: 100 } }
+    let(:income_calculator) { double(calculate: income_calculation_result) }
+
     before do
-      allow(subject).to receive(:valid?).and_return(true)
-      allow(subject).to receive(:persist!)
-      allow(EvidenceCheck).to receive(:find)
-      allow_any_instance_of(EvidenceCheck).to receive(:update)
+      allow(IncomeCalculation).to receive(:new).with(evidence.application, 500).and_return(income_calculator)
+      evidence.update_attributes(params)
+    end
+
+    subject { form.save }
+
+    it 'saves the income on the evidence in the correct format' do
+      subject && evidence.reload
+
+      expect(evidence.income).to eql(500)
     end
 
-    it 'saves the form data into appropriate models' do
-      expect(subject.save).to eq true
+    it 'saves the income on the income calculation outputs' do
+      subject && evidence.reload
```

Comment:
Could the code on the lines 76 & 70 be put in a

```ruby
before(:each) { }
```

Not a show stopper, just saw it and figured the repetition could potentially be removed.

If this is the only thing in this PR, feel free to merge it and maybe revisit this on a next PR...